### PR TITLE
Add a CanDeployOnRamps property to DeployToUpgrade

### PR DESCRIPTION
**Pull request by penev92**
So stuff like the Mobile Sensor Array can't deploy on ramps (slopes). (As seen in @SoScared's recent TS stream).

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs [hunk 1]
```diff
@@ -143,6 +146,11 @@ public void ResolveOrder(Actor self, Order order)
 
 		bool OnValidTerrain()
 		{
+			return CheckTerrain() && CheckRamps();
+		}
+		
+		bool CheckTerrain()
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs]: `OnValidTerrainType` would be more descriptive.

File: OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs [hunk 2]
```diff
@@ -156,6 +164,23 @@ bool OnValidTerrain()
 			return info.AllowedTerrainTypes.Contains(terrainType);
 		}
 
+		bool CheckRamps()
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs]: `OnValidRampType` would be more descriptive.

**Comment by abcdefg30** [comment]
Looks good to me. :+1: / :white_check_mark:

File: OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs [hunk 3]
```diff
@@ -28,6 +28,9 @@ public class DeployToUpgradeInfo : ITraitInfo, Requires<UpgradeManagerInfo>
 			"Leave empty to allow any.")]
 		public readonly HashSet<string> AllowedTerrainTypes = new HashSet<string>();
 
+		[Desc("Can this actor deploy on slopes?")]
+		public readonly bool CanDeployOnRamps = false;
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs]: The description and field name are inconsistent.

Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs]: It is just not redundant which is actually better. Explaining terms with the same word isn't helping.

**Comment by Mailaender** [comment]
Works as promised. :white_check_mark:

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/6dbb6a0ce3301ce0e346b2ba1881be09e264d4e2...0b2ca66206d1851a474e213af9178766a6037e86)

**Pull request merged**